src/ui/components/ShowMoreCard/index.js

```diff
@@ -1,55 +1,126 @@
+/* @flow */
 /* eslint-disable react/no-danger */
-
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
 import { compose } from 'redux';
 
 import translate from 'core/i18n/translate';
+import withUIState from 'core/withUIState';
 import { sanitizeHTML } from 'core/utils';
 import Card from 'ui/components/Card';
+import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
 // This refers to height of `Card-contents` CSS class,
 // beyond which it will add read more link.
 export const MAX_HEIGHT = 150;
 
-export class ShowMoreCardBase extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    header: PropTypes.node,
-    i18n: PropTypes.object.isRequired,
-  };
+type UIStateType = {|
+  showAllContent: boolean,
+  readMoreExpanded: boolean,
+|};
+
+type Props = {|
+  children: React.Element<any> | string,
+  className?: string,
+  header?: React.Element<any> | string,
+  id: string,
+|};
+
+type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+  children: React.Element<any>,
```

Comment:
I remove this here flow complains because of the following:
```javascript
let html =
      children.props &&
      children.props.dangerouslySetInnerHTML &&
      children.props.dangerouslySetInnerHTML.__html;
...
```
it seems weird to have this in both spots (in Props too), but I guess this is okay?